tools/dist/promote/_promote.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+if [ -z ${srcdir+x} ]; then
+  echo "\$srcdir is not set"
+  exit 1
+fi
+
+if [ -z ${dstdir+x} ]; then
+  echo "\$dstdir is not set"
+  exit 1
+fi
+
+version=$1
+
+for subdir in $(cd $srcdir && ls); do
+
+  if [ -d "${srcdir}/${subdir}" ]; then
+    resha=no
+
+    if [ "X${version}" != "X" ] && [ "X${version}" != "X${subdir}" ]; then
+      continue
+    fi
+
+    for donefile in $(cd ${srcdir}/${subdir} && ls *.done 2> /dev/null); do
+      doneref=$(echo $donefile | sed 's/\.done$//')
+
+      if [[ -f "${srcdir}/${subdir}/${doneref}" || -d "${srcdir}/${subdir}/${doneref}" ]]; then
+        echo "Promoting ${subdir}/${doneref}..."
+        mkdir -p "${dstdir}/${subdir}"
+        rm -rf "${dstdir}/${subdir}/${doneref}"
+        cp -dr "${srcdir}/${subdir}/${doneref}" "${dstdir}/${subdir}/${doneref}"
```

Comment:
I think the main difference between `-a` and `-dr` is that `-d` only does a `--preserve=links` while `-a` does a `--preserve=all` and the implication here is that the mode isn't copied - which is 664/775 for staging but ends up being 644/755 for dist. _but_ that doesn't really matter because there aren't any other users in the dist group.. So I'm still not sure what I was thinking! Will `-a` and see what breaks, if anything.